src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.SingleStatementCodeGenerator.cs

```diff
@@ -28,38 +25,15 @@ public sealed class SingleStatementCodeGenerator(
                 CSharpCodeGenerationOptions options,
                 bool localFunction) : CSharpCodeGenerator(insertionPoint, selectionResult, analyzerResult, options, localFunction)
             {
-                public static bool IsExtractMethodOnSingleStatement(SelectionResult code)
-                {
-                    var result = (CSharpSelectionResult)code;
-                    var firstStatement = result.GetFirstStatement();
-                    var lastStatement = result.GetLastStatement();
-
-                    return firstStatement == lastStatement || firstStatement.Span.Contains(lastStatement.Span);
-                }
-
                 protected override SyntaxToken CreateMethodName() => GenerateMethodNameForStatementGenerators();
 
                 protected override ImmutableArray<StatementSyntax> GetInitialStatementsForMethodDefinitions()
                 {
-                    Contract.ThrowIfFalse(IsExtractMethodOnSingleStatement(CSharpSelectionResult));
+                    Contract.ThrowIfFalse(CSharpSelectionResult.IsExtractMethodOnSingleStatement());
 
                     return ImmutableArray.Create(CSharpSelectionResult.GetFirstStatement());
                 }
 
-                protected override SyntaxNode GetOutermostCallSiteContainerToProcess(CancellationToken cancellationToken)
-                {
-                    var callSiteContainer = GetCallSiteContainerFromOutermostMoveInVariable(cancellationToken);
-                    if (callSiteContainer != null)
-                    {
-                        return callSiteContainer;
-                    }
-                    else
```

Comment:
top half of this moved up to base type.  bottom half to the 'selection result' type.